base/all-all/sbardef.lmp

```diff
@@ -123,7 +301,117 @@
               "type": 1,
               "param": 0,
               "maxlength": 3,
-              "conditions": null,
+              "conditions":
+              [
+                {
+                  "condition": 22,
+                  "param": 0
+                }
+              ],
+              "children": null
+            }
+          },
+          {
+            "percent":
+            {
+              "x": 235,
+              "y": 3,
+              "alignment": 2,
+              "font": "BigRed",
+              "tranmap": null,
+              "translation": "CRRED",
+              "type": 1,
+              "param": 0,
+              "maxlength": 3,
+              "conditions":
+              [
+                {
+                  "condition": 21,
+                  "param": 0
+                },
+                {
+                  "condition": 22,
+                  "param": 1
+                }
+              ],
+              "children": null
+            }
+          },
+          {
+            "percent":
+            {
+              "x": 235,
+              "y": 3,
+              "alignment": 2,
+              "font": "BigRed",
+              "tranmap": null,
+              "translation": "CRGOLD",
+              "type": 1,
+              "param": 0,
+              "maxlength": 3,
+              "conditions":
+              [
+                {
+                  "condition": 21,
+                  "param": 25
+                },
+                {
+                  "condition": 22,
+                  "param": 1
+                }
+              ],
+              "children": null
+            }
+          },
+          {
+            "percent":
+            {
+              "x": 235,
+              "y": 3,
+              "alignment": 2,
+              "font": "BigRed",
+              "tranmap": null,
+              "translation": "CRGREEN",
+              "type": 1,
+              "param": 0,
+              "maxlength": 3,
+              "conditions":
+              [
+                {
+                  "condition": 21,
+                  "param": 50
+                },
+                {
+                  "condition": 22,
+                  "param": 1
+                }
+              ],
+              "children": null
+            }
+          },
+          {
+            "percent":
+            {
+              "x": 235,
+              "y": 3,
+              "alignment": 2,
+              "font": "BigRed",
+              "tranmap": null,
+              "translation": "CRBLUE",
+              "type": 1,
+              "param": 0,
+              "maxlength": 3,
+              "conditions":
+              [
+                {
+                  "condition": 21,
+                  "param": 100
+                },
+                {
+                  "condition": 22,
+                  "param": 1
+                }
+              ],
```

Comment:
@fabiangreffrath I implement Boom colors by extending SBARDEF. We will have to increase the lump version, and it will not work with the default Legacy of Rust WAD. So users who want this feature will have to overwite SBARDEF.

Reply:
> I implement Boom colors by extending SBARDEF.

Thank you! Doesn't seem to work with my build, though. 



> and it will not work with the default Legacy of Rust WAD. So users who want this feature will have to overwite SBARDEF.

I find that reasonable, since Legacy of Rust decided to ship its own HUD definitions, and we ship ours. 

> We will have to increase the lump version

Should we introduce a Woof namespace probably?

Reply:
> > I implement Boom colors by extending SBARDEF.
> 
> Thank you! Doesn't seem to work with my build, though.

Is your woof.pk3 updated (it should contain sbardef.lmp). Did you enable “Colored Numbers” in the menu?

Reply:
I think so:

https://github.com/user-attachments/assets/56a8a65a-0c63-4393-a8e7-a4678fdab612

Reply:
> I think so:

But it works, so far I've only done it for the classic status bar 😄

What do you think about this approach? I want to remove the color configuration options (except `sts_colored_numbers`), users can edit SBARDEF themselves. But editing it is not very fun.

Reply:
> But it works, so far I've only done it for the classic status bar 😄

Ah, this explains a lot! :grin: 

> What do you think about this approach? I want to remove the color configuration options (except `sts_colored_numbers`), users can edit SBARDEF themselves. But editing it is not very fun.

I think it's reasonable. I don't think I have ever edited the threshold values in my entire life, but I'm not sure if others ever did.

However, we still need the health-to-color logic for the health-colored crosshair, so maybe it's still better to leave this in a function?

Another question: If we have, say, 200 health, do we draw the health widget four times? I mean, once in red, because 200 >= 0, once in yellow, because 200 >= 25, once in green, because 200 >= 50 and finally in blue, because 200 >= 100?

Reply:
Also, health and armor "turn to stone" (i.e. get drawn in gray) in invincibility mode. We shouldn't introduce too many regressions for people who use the colored status bar feature.

Reply:
> However, we still need the health-to-color logic for the health-colored crosshair, so maybe it's still better to leave this in a function?

Sure, we'll keep this function.

> Another question: If we have, say, 200 health, do we draw the health widget four times? I mean, once in red, because 200 >= 0, once in yellow, because 200 >= 25, once in green, because 200 >= 50 and finally in blue, because 200 >= 100?

Yes 😆 I need to add additional conditions (`>=` and `<=`).